reporter/src/main/kotlin/reporters/AbstractNoticeReporter.kt

```diff
@@ -0,0 +1,94 @@
+/*
+ * Copyright (c) 2017-2018 HERE Europe B.V.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ *
+ * SPDX-License-Identifier: Apache-2.0
+ * License-Filename: LICENSE
+ */
+
+package com.here.ort.reporter.reporters
+
+import ch.frankel.slf4k.info
+
+import com.here.ort.model.OrtResult
+import com.here.ort.reporter.Reporter
+import com.here.ort.reporter.ResolutionProvider
+import com.here.ort.utils.log
+import com.here.ort.utils.spdx.getLicenseText
+
+import java.io.File
+import java.util.SortedMap
+import java.util.SortedSet
+
+abstract class AbstractNoticeReporter : Reporter() {
+    override fun generateReport(ortResult: OrtResult, resolutionProvider: ResolutionProvider, outputDir: File): File? {
+        require(ortResult.scanner != null) {
```

Comment:
please use requireNotNull

https://kotlinlang.org/api/latest/jvm/stdlib/kotlin/require-not-null.html

Reply:
Will do in a separate commit as this is just moved code.